test/hotspot/jtreg/vmTestbase/nsk/share/jdi/JDIBase.java

```diff
@@ -164,4 +166,46 @@ protected void breakpointForCommunication() throws JDITestRuntimeException {
         throw new JDITestRuntimeException("** event '" + event + "' IS NOT a breakpoint **");
     }
 
+    // Waiting for ThreadStart and ThreadDeath event we can get events from system threads,
+    // so need to ensure we get the event from the desired thread.
+    protected void waitThreadStart(String threadName) throws JDITestRuntimeException {
+        while (true) {
+            getEventSet();
+            Event event = eventIterator.nextEvent();
+            if (event instanceof ThreadStartEvent) {
+                ThreadStartEvent evt = (ThreadStartEvent)event;
+                if (evt.thread().name().equals(threadName)) {
+                    break;
+                }
+            } else if (!(event instanceof ThreadDeathEvent)) {
+                // unexpected event. let the test handle it
+                break;
+            }
+            // log the event and wait for next one
+            log2("Waiting for ThreadStartEvent, got " + event);
```

Comment:
This message is misleading. It could in fact be a ThreadStart, but for some other thread other than the named thread. I think the code would be clearer if there was no fall-through case. Put this code in an else, but also include a different message as part of the else for the `if (evt.thread().name().equals(threadName))` section. Maybe say something like "...got ThreadStartEvent for wrong thread".